docs/source/development/automated_testing.md

```diff
@@ -0,0 +1,165 @@
+# Automated Testing
+
+An important step towards achieving high code quality and maintainability in your Kedro project is the use of automated tests. Let's look at how you can set this up.
+## Introduction
+
+Software testing is the process of checking that the code you have written fulfills its requirements. Software testing can either be **manual** or **automated**. In the context of Kedro:
+- **Manual testing** is when you run part or all of your project and check that the results are what you expect.
+- **Automated testing** is writing new code (using libraries called _testing frameworks_) that runs part or all of your project and automatically checks the results against what you expect.
+
+As a project grows larger, new code will increasingly rely on existing code. As these interdependencies grow, making changes in one part of the code base can unexpectedly break the intended functionality in another part.
+
+The major disadvantage of manual testing is that it is time-consuming. Manual tests are usually run once, directly after new functionality has been added. It is impractical to repeat the manual tests for the entire existing code base each time a change is made, which means this strategy often misses breaking changes. 
+
+The solution to this problem is automated testing, which allows many tests across the whole code base to be run in seconds, every time a new feature is added or an old one is changed. In this way, breaking changes can be discovered during development rather than in production.
+
+## Set up automated testing with `pytest`
+
+There are many testing frameworks available for Python. One of the most popular is [`pytest`](https://docs.pytest.org/). `pytest` is often used in Python projects for its short, readable tests and powerful set of features.
+
+Let's look at how you can start working with `pytest` in your Kedro project.
+
+### Install `pytest`
+
+Install `pytest` as you would install other packages with `pip`, making sure your project's virtual environment is active.
+
+```bash
+pip install pytest
+```
+
+### Create a `/tests` directory
+
+Now that `pytest` is installed, you will need a place to put your tests. Create a `/tests` folder in the `/src` directory of your project.
+
+```bash
+mkdir <project_root>/src/tests
+```
+
+### Test directory structure
+
+The subdirectories in your project's `/tests` directory should mirror the directory structure of your project's `/src/<project_name>` directory. All files in the `/tests` folder should be named `test_<file_being_tested>.py`. For instance:
```

Comment:
```suggestion
The subdirectories in your project's `/tests` directory should mirror the directory structure of your project's `/src/<package_name>` directory. All files in the `/tests` folder should be named `test_<file_being_tested>.py`. For instance:
```

We should stick with this naming convention.  https://github.com/kedro-org/kedro/pull/1753

Reply:
You're right about this. I changed three instances of <project_name> to <package_name> in [6a6a278](https://github.com/kedro-org/kedro/pull/1879/commits/6a6a2788d43e93c554e0e2381636df0ff1d366d7).